### PR TITLE
Improving my abortive, awful research vault (work in progress)

### PR DESCRIPTION
**Pull request by Arthurdentist**
A long time ago I started work on a big vault. The idea was that the vault would be some kind of research facility that had been owned in secret by Central Command, until the experiments escaped and killed the personnel inside. Unfortunately this vault was never finished, and because I am a lazy bastard it was eventually abandoned by me. That original PR was #8968

The PR was eventually resurrected by @Shadowmech88 here #13540, and the vault was merged.

I didn't actually realize the vault existed in game for the longest time, until I found it  by accident and got a weird Deja Vu feeling, so now I've decided to return and finish what I've started. The vault has a few problems that I think need addressing, so yeah here we go.

## The vault as it exists now
current vault

## The changes I have made so far
new vault

To begin with I have changed the mobs that can spawn in the vault, and removed the cyberhorrors entirely. Now the vault will spawn different levels of zombies, although I have increased the frequency of them spawning a fair bit to compensate. This was because a lot of people have said that the current vault is extremely unforgiving, without really having anything in it that would be worth acquiring.

I've also added some new lootspawners to the vault. The main issue with the loot as it was, is that the rewards you could get mostly came in the form of weapons (to get the weapons you would have already have to have weapons to kill the cyber horrors) or surgery / science items (its very unlikely you'll find someone to do advanced lung surgery on you in the middle of space). A lot of that loot remains for thematic reasons but now you also have a decent chance of getting some other good stuff, including favorable genetic mutations which should hopefully give some more incentive to explore the facility. 

I've also added a destination disk for the vault. This disk cannot be generated in spessmart (as far as I know) but you can find it very early on in the vault. This ties into my other changes. Although I haven't finished it yet, I thought it would be cool if this facility could be cleared out and used as a viable base. I've added some lights and a broken APC which should (eventually) be fixable, meaning that if you can clear out the zombies you will be able to turn this facility into a fully functioning and quite powerful base. 

There is a lot more work that needs to be done, and I need to figure out some mapping shite as well as figuring out what sort of power sources a facility like this should have, but if anyone has any suggestions feel free to share them here or in discord. 

TL;DR: Admin changes old vault

### CODE REVIEW

**Comment by MadmanMartian** [comment]
From testing, the lower scanner bed in the medical bay blocks entry and exit from that doorway. Is this intended?

**Comment by Arthurdentist** [comment]
@MadmanMartian not intended, guessing thats a problem with the current vault as it exists. I will remove the scanner entirely because I can't really fit the whole thing in, its also missing the actual screen. 

The old vault was actually like the second vault I had ever attempt to map, so there are a lot of small issues like that which need to be resolved.

**Comment by MadmanMartian** [comment]
The screen is auto-generated on load IIRC from the debug logs, and is what actually blocks the door.

**Comment by Arthurdentist** [comment]
best thing would be to swap it with the bed immediately above it, that should give people room to walk around it.

**Comment by Blithering** [comment]
Note on this vault: due to the security lockers, the SPSes in them will show the exact location of the vault to anyone who has a SPS

**Comment by Arthurdentist** [comment]
do you think that should be changed? I know you can locate this vault already, but if you think it'd be better to prevent that, then I can remove it.

**Comment by Blithering** [comment]
I'm neutral on it, just thought it may have been unintentional. Sec rarely has time to go vaulthunting

**Comment by ghost** [comment]
I find it very curious for you to say these things given I consider it one of the best vaults in terms of loot and easiest to clear.  Oddly enough I don't think you consider its single best loot spawn to even make the radar.  Oh well.  The changes are solid.  Always nice to see space stuff touched.

**Comment by Arthurdentist** [comment]
@HarseTheef what do you consider its single best loot spawn? the ancient spacesuit?

**Comment by Kurfursten** [comment]
In the original design because of the hostile placement it was possible to get just the meat cleaver and leave without ever fighting anything. Dunno the status now.

>Another vault dest disk

We should seriously consider mapping a civilian shuttle in arrivals that's public access. These disks are currently only useful to miners and scientists, and even bringing a friend is somewhat annoying because of the fast timing on external airlocks. It's a big hassle and lowers the value of disks.

**Comment by Arthurdentist** [comment]
@Kurfursten adding a general purpose civilian ship is well byond my capabilities but if someone wanted to do that then it would be a good idea.

**Comment by ghost** [comment]
the 25u medinanos sitting there in a beaker next to a free chemmaster3k and pill bottle

**Comment by Arthurdentist** [comment]
@HarseTheef Do you think that should be removed or replaced? I mean this is why I threw this up as early as I did, I am looking for concrete suggestions and ideas.

**Comment by ghost** [comment]
I actually like it.  It is very unique in terms of vaults and goes well with the theme of it being a medical research outpost.  If I were you, I would move away from weapons.  There are more than enough of those in space.  Keep it with the theme of rare medical supplies/chemicals and things that would be researched.  Possibly something like a hivelord in a cell.  Maybe an npc changeling horror or a necromorph somewhere.  I just realized that we need cool changeling npcs.

**Comment by Arthurdentist** [comment]
The loot spawners I added do have a chance of spawning laser guns, but most of the time it will spawn either a DNAinjector or some medical supplies, it can spawn a smart laser gun but its a 1 in 21 chance that any spot which actually spawns loot (80% chance a loot spawner spawns anything) will spawn a smart laser. 

As for the mobs, instead of cyber horrors each mob spawner can spawn skeletons, zombies, zombie ghouls or even a glowing one, with zombies being the most likely. Mob spawners are also more likely to actually spawn mobs at 60% instead of 50%. The fun thing about zombie behaviour is that they will move to eat corpses when they see them, so if for instance a zombie spawns in one of the cells, they smash the plasma glass to get at the dead corpse in the top-middle cell. This helps to make the vault look a bit more rundown and also makes the zombie placement a bit less predictable. 

One thing I've always considered is whether I should maybe add a 'boss' mob to the vault, something unique that spawns in one of the rooms and has unique drop gear as well. Perhaps a zombie surgeon of some kind? I will have to consider that.

**Comment by ghost** [comment]
sounds rad enough.  I heavily support a boss mob.

**Comment by ShiftyRail** [comment]
It's a lot of effort, but you could add solars panels at the rear of the vault
Even if they are placebo and not connected to anything, it's good for the immersions

**Comment by Blithering** [comment]
Or a SMES in the empty room with the wire leading to it on the top left

**Comment by SkowronX** [comment]
make a different sprite set

**Comment by Blithering** [comment]
Sprite set for what now?

**Comment by SkowronX** [comment]
for the vault.

**Comment by Kurfursten** [comment]
You mean the vault style airlocks?

**Comment by SkowronX** [comment]
That too. I'd just like vaults to have a unique look to themselves

**Comment by Arthurdentist** [comment]
Started work on adding a 'mini-boss' to the vault. Going to need a lot more work and I will provide some more info on it soon.

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	corpse = null //he isn't dead yet
+
+	faction = "skeleton" //don't want him to get killed by skeletons before his transformation
+
+/mob/living/simple_animal/hostile/humanoid/surgeon/normal/Die()
+	src.say("[pick("YOU CAN'T KILL ME THAT EASILY!", "I WONT LET YOU STOP ME!", "GET OUT OF MY FACILITY!", "I MUST CONTINUE MY RESEARCH!", "I'M GONNA WRECK IT!", "I'VE GOT A BONE TO PICK WITH YOU!")]")
+	var/obj/effect/effect/smoke/S = new /obj/effect/effect/smoke(get_turf(src))
+	to_chat("<span class=danger><B>Before he can die, the mad surgeon takes a drink of a foul-smelling concoction and begins to mutate! </span></B>")
+	S.time_to_live = 20
+	src.
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: I've seen some src in my day but this has got to be the srciest

Comment (Arthurdentist) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: its also nowhere near finished (as I said)

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+	..()
+	return qdel(src)
+
+/mob/living/simple_animal/hostile/humanoid/surgeon/skeleton
+	name = "Skeletal Strongman"
+	desc = "He wont be pushed around any longer"
+	//icon_state = "skeletonboss"
+	//icon_living = "skeletonboss"
+	maxHealth = 1000
```
Comment (MadmanMartian) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: to put this number into context, the captains laser (biggest damage dealer that can be acquired through legitimate means) deals about 40 damage per shot.

Didn't realise this was a JRPG

Comment (Arthurdentist) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: Its health at 1000 is equal to Mecha Hitler, double what a spider queen has and four times as much as a cult juggernaut. The aim is to have a big, hulking skeleton, that hits hard but moves slowly. Its also not even functioning yet and no numbers are final. 

We really haven't reached JRPG levels yet.

Comment (Kurfursten) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: I tend to agree with Martian. Even on station where there are rechargers onhand, spider queens can feel almost comical in how many shots they take. I remember one showing up on the Packed bridge the other night. It couldn't figure out how to get around the smashed monitors and firelocks fortunately so I just sat there for a good two or three minutes repeatedly recharging and shooting it. It wasn't very engaging.

Comment (MadmanMartian) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: We need to fix up the mob AI. If this is a boss mob that can only occur in one place then that's good. Make sure it can't occur through the masks, gold slime cores, antagonist pokeballs, etc.

Comment (Arthurdentist) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: I said more info would come, I only put this up for now so that I could save a bit of progress.

He wont appear anywhere outside of his one little space, that is the intention.

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 3]
```diff
@@ -20,50 +19,53 @@
 	//corpse = /obj/effect/landmark/corpse/surgeon
 	items_to_drop = list(/obj/item/weapon/circular_saw/plasmasaw)
 
-/mob/living/simple_animal/hostile/humanoid/surgeon/normal
-	name = "Doctor Sawyer"
+/mob/living/simple_animal/hostile/humanoid/surgeon/boss
+	name = "Doctor Placeholder" //I can't think of a good name
```
Comment (Probe1) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: How about Zizek

Comment (Arthurdentist) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: space zizek

**Comment by Arthurdentist** [comment]
Deep in the vault you can now find one remaining member of the facility. Stuck in space surrounded by zombies has driven him quite mad, but that hasn't stopped him from working on his research.

You can kill him for some unique loot (work in progress), but he wont go down without a fight, and not before showing you the culmination of all his research.

**Comment by Arthurdentist** [comment]
beautiful sprites courtesy of @Blithering as well 
surgeon

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm (hunk 3 above)
Comment (Probe1) [reply code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: I'm sold.

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	melee_damage_upper = 20
+
+	corpse = null //he isn't dead yet
+
+	faction = "necro" //don't want him to get killed by skeletons before his transformation
+
+/mob/living/simple_animal/hostile/humanoid/surgeon/boss/Die()
+	visible_message("<span class=danger><B>Before he can die, the mad surgeon takes a drink of a foul-smelling concoction and begins to mutate! </span></B>")
+	say("[pick("YOU CAN'T KILL ME THAT EASILY!", "I WONT LET YOU STOP ME!", "GET OUT OF MY FACILITY!", "I MUST CONTINUE MY RESEARCH!", "I'M GONNA WRECK IT!", "I'VE GOT A BONE TO PICK WITH YOU!")]")
+	var/obj/effect/effect/smoke/S = new /obj/effect/effect/smoke(get_turf(src))
```
Comment (MadmanMartian) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm:40]: this will only make a single bit of smoke, are you sure you don't want it to spread?

If you do, look at /datum/effect/effect/system/smoke_spread

**Comment by Arthurdentist** [comment]
updated vault

made additional changes to the vault. Extended it out a bit and added an AME room, it makes sense as a power source and it would also be something an intrepid explorer could put together.

Also added two internal bleeding first aid kits, one in the boss room and one in the medical supply room. This is because this vault can spawn a lot of mobs to fight, and there is a decent chance you could get internal bleeding which is seriously no fun allowed deep in space. This should help out a bit.

Also removed glowing ones as a possibility in this vault, mainly cause their corpses can cause radiation and most space explorers will lack any decent means of dealing with this. I also thought they didn't fit with the vault thematically.

**Comment by Arthurdentist** [comment]
finaliternation

Pretty happy with the map changes now, this will be the final iteration unless there is something really gamebreaking or bad.

Resolved an issue with the APC and did some other stuff. Everything should be operational now.

**Review by MadmanMartian — APPROVED** [review]

**Comment by MadmanMartian** [comment]
I thought this wasn't done

**Comment by Arthurdentist** [comment]
its not

**Comment by Caboozles** [comment]
So I'm not sure if this is the right place to dump this, but I had a commander spawn at the front door, essentially giving me all-access back on the station due to his ID. I'll try to replicate it next time I play spaceman, but I'm not sure if an all-access ID was an intended part of the new loot system you changed.

File: maps/randomvaults/research_facility.dmm [hunk 5]
```diff
@@ -1,185 +1,241 @@
...
+"dy" = (/obj/abstract/map/spawner/misc/medivault,/turf/simulated/floor{blocks_air = 1; icon_state = "dark"},/area/vault/research)
+"dz" = (/obj/item/weapon/storage/box/emps,/turf/simulated/floor{blocks_air = 1; icon_state = "dark"},/area/vault/research)
+"dA" = (/obj/machinery/door/airlock/maintenance,/turf/simulated/floor{blocks_air = 1; icon_state = "dark"},/area/vault/research)
+"dB" = (/obj/machinery/door/airlock/research,/turf/simulated/floor/carpet,/area/vault/research)
+"dC" = (/obj/abstract/map/spawner/mobs/medivault,/turf/simulated/floor/carpet,/area/vault/research)
+"dD" = (/obj/item/weapon/storage/pill_bottle/dice,/turf/simulated/floor/carpet,/area/vault/research)
+"dE" = (/obj/structure/bed,/obj/item/weapon/bedsheet/green,/obj/machinery/light/small{dir = 4},/turf/simulated/floor/carpet,/area/vault/research)
+"dF" = (/obj/machinery/door/window/plasma,/turf/simulated/floor/wood,/area/vault/research)
+"dG" = (/turf/simulated/floor/carpet,/area/vault/research)
+"dH" = (/obj/item/weapon/stool,/obj/effect/landmark/corpse/commander,/turf/simulated/floor/carpet,/area/vault/research)
```
Comment (MadmanMartian) [inline maps/randomvaults/research_facility.dmm:190]: Commander has an all access ID. This needs sorting ASAP
